Tools/CMake/AMReXInstallHelpers.cmake

```
@@ -33,6 +33,13 @@ function ( generate_amrex_config_header )
       set(COMPILER_ID_MACRO  __PGI)
    elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "Clang" )
       set(COMPILER_ID_MACRO  __llvm__)
+   elseif ("${CMAKE_CXX_COMPILER_ID}" STREQUAL "AppleClang" )
+      set(COMPILER_ID_MACRO  __llvm__)
+   else ()
+      # avoid malforming the AMReX_Config.H file by setting something
+      set(COMPILER_ID_MACRO  ${CMAKE_CXX_COMPILER_ID})
+      message(WARNING "Compiler '${CMAKE_CXX_COMPILER_ID}' is not tested by AMReX developers! "
+                      "Here be dragons.")
```

I guess if we fallback to guessing `__llvm__` for new compilers we are 90% of the time right (DPC++, HIP, other SYCL compilers, ...).

Shall I set accordingly?
```suggestion
      set(COMPILER_ID_MACRO __llvm__)
      message(WARNING "Compiler '${CMAKE_CXX_COMPILER_ID}' is not tested by AMReX developers! "
                      "Assuming some kind of LLVM fork...")
```

---

@ax3l I don't think Line 40 will work. For example, if lines 36-37 weren't present then for AppleClang this would end up as

```C++
#ifndef AppleClang
static_assert(false,"libamrex was built with AppleClang");
#endif
```

---

Oh that's right, thanks!

Hm, it might be then time of a nice massive list of IDs as defined in https://sourceforge.net/p/predef/wiki/Compilers/

I have to admit, the check here is a bit of snakeoil anyway:
- it is overly strict: some compilers are intentionally ABI compatible, e.g. with a system GCC
- it is not safe: a major version in the same compiler can still create ABI problems
- it is incomplete: as described above; we have to update AMReX for every new compiler users want to use

It's a bit of a workflow/policy decision what we want to do here...